pennylane/labs/resource_estimation/resource_container.py

```diff
@@ -238,6 +245,78 @@ def mul_in_parallel(first: Resources, scalar: int, in_place=False) -> Resources:
     return Resources(new_wires, new_gates, new_gate_types)
 
 
+def substitute(
+    initial_resources: Resources, name: str, replacement_resources: Resources, in_place=False
+) -> Resources:
+    """Replaces a specified gate in a :class:`~.resource.Resources` object with the contents of another :class:`~.resource.Resources` object.
+
+    Args:
+        initial_resources (Resources): the resources to be modified
+        name (str): the name of the operation to be replaced
+        replacement (Resources): the resources to be substituted instead of the gate
+        in_place (bool): determines if the initial resources are modified in place or if a new copy is created
+
+    Returns:
+        Resources: the updated :class:`~.Resources` after substitution
+
+    .. details::
+
+        **Example**
+
+        In this example we replace the resources for the :code:`RX` gate. First we build the :class:`~.Resources`:
+
+        .. code-block:: python3
+
+            from pennylane.labs.resource_estimation import Resources
+
+            replace_gate_name = "RX"
+
+            initial_resources = Resources(
+                num_wires = 2,
+                num_gates = 3,
+                gate_types = {"RX": 2, "CNOT": 1},
+            )
+
+            replacement_rx_resources = Resources(
+                num_wires = 1,
+                num_gates = 7,
+                gate_types = {"Hadamard": 3, "S": 4},
+            )
+
+        Executing the substitution produces:
+
+        >>> res = qml.labs.resource_estimation.substitute(
+        ...     initial_resources, replace_gate_name, replacement_rx_resources,
+        ... )
+        >>> print(res)
+        wires: 2
+        gates: 15
+        gate_types:
+        {'CNOT': 1, 'Hadamard': 6, 'S': 8}
```

Comment:
```suggestion
        >>> from pennylane.labs.resource_estimation import substitute
        >>> substitute(
        ...     initial_resources, replace_gate_name, replacement_rx_resources,
        ... )
        >>> print(res)
        wires: 2
        gates: 15
        gate_types:
        {'CNOT': 1, 'Hadamard': 6, 'S': 8}
```

Reply:
labs will not be accessible from inside pennylane when its imported. Rather we need to access it through the file structure using `from pennylane.labs import ...`